group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -426,9 +429,43 @@ public CompletableFuture<ListGroupsResponseData> listGroups(
             return FutureUtils.failedFuture(Errors.COORDINATOR_NOT_AVAILABLE.exception());
         }
 
-        return FutureUtils.failedFuture(Errors.UNSUPPORTED_VERSION.exception(
-            "This API is not implemented yet."
-        ));
+        List<CompletableFuture<ListGroupsResponseData>> futures = new java.util.ArrayList<>(Collections.emptyList());
+        for (int i = 0; i < numPartitions; i++) {
```

Comment:
This seems to be inefficient because the coordinator may not be responsible for all the partitions. I thought that we could use `CoordinatorRuntime#partitions` to get the list of registered partitions. Have you considered this?

Reply:
Thanks for the reminder, I tried to find it but didn't know it, so I chose an inefficient implementation.